kategory-effects-rx2/src/main/kotlin/kategory/effects/data/FlowableKW.kt

```diff
@@ -26,6 +32,22 @@ data class FlowableKW<A>(val flowable: Flowable<A>) : FlowableKWKind<A> {
     fun <B> switchMap(f: (A) -> FlowableKWKind<B>): FlowableKW<B> =
             flowable.switchMap { f(it).ev().flowable }.k()
 
+    fun <B> foldL(b: B, f: (B, A) -> B): B = flowable.reduce(b, f).blockingGet()
+
+    fun <B> foldR(lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> {
+        fun loop(fa_p: FlowableKW<A>): Eval<B> = when {
```

Comment:
Does the loop need to be tailrec?

Reply:
foldR written with `Eval` should be stack safe, right?

Reply:
You're right, we have the same impl for ListKW, MapKW etc :P silly me